### PR TITLE
feat: draw clipart option in drawer

### LINKED ISSUE #385: draw badge option in navigation drawer
**Describe the feature**
Include Draw Badge Function in Navigation Drawer

**Would you like to work on the issue?**
YES

### PR DESCRIPTION
**Pull request by adityastic**
Fixes #385

Screenshots:
On Drawer:

After Opening:

Drawn and Saved:

ClipArts get Updated:

### CODE REVIEW

**Comment by adityastic** [comment]
@mariobehling @yashk2000 @iamareebjamal @liveHarshit @anhanh11001 review

**Comment by mariobehling** [comment]
Awesome! Please resolve conflicts.

**Comment by mariobehling** [comment]
@adityastic Please add a "Draw Clipart Icon" at the end of the clipart list. This button should direct to the "Draw Clipart screen.

**Comment by adityastic** [comment]
@mariobehling Updated

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt [hunk 1]
```diff
@@ -378,7 +379,11 @@ class TextArtFragment : BaseFragment() {
     private fun putBitmapInEditText(drawableInfo: DrawableInfo) {
         val strToAppend = "«${drawableInfo.id}»"
         val spanStringBuilder = SpannableStringBuilder(strToAppend)
-        spanStringBuilder.setSpan(CenteredImageSpan(requireContext(), ImageUtils.trim(ImageUtils.vectorToBitmap(drawableInfo.image as VectorDrawable), 70)), 0, strToAppend.length, 33)
+        spanStringBuilder.setSpan(CenteredImageSpan(requireContext(), ImageUtils.trim(
+            if (drawableInfo.image is VectorDrawable)
+                ImageUtils.vectorToBitmap(drawableInfo.image)
+            else
+                (drawableInfo.image as BitmapDrawable).bitmap, 70)), 0, strToAppend.length, 33)
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt]: use smart cast instead: `drawableInfo.image is BitmapDrawable`

File: app/src/main/res/layout/fragment_draw.xml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="match_parent"
+                android:layout_margin="10dp"
+                android:gravity="center"
+                android:onClick="@{() -> viewModel.changeEraseState()}"
+                android:orientation="vertical">
+
+                <ImageView
+                    android:layout_width="48dp"
+                    android:layout_height="48dp"
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_draw.xml]: Dimensions should define inside dimentsion reqources file.

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/DrawFragment.kt [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+class DrawFragment : BaseFragment() {
+
+    companion object {
+        @JvmStatic
+        fun newInstance() =
+            DrawFragment()
+    }
+
+    private val viewModel by viewModel<DrawViewModel>()
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/badgemagic/ui/fragments/DrawFragment.kt]: Too generic, it can be `drawViewModel`

**Comment by adityastic** [comment]
@liveHarshit updated

**Comment by adityastic** [comment]
@mariobehling here is the update. Screenshot of the update: 
385add

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt (hunk 1 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt]: There are 3 more unsafe casts like that.

File: app/src/main/res/layout/fragment_draw.xml (hunk 2 above)
Comment (liveHarshit) [reply app/src/main/res/layout/fragment_draw.xml]: There are more than one.

Comment (adityastic) [reply app/src/main/res/layout/fragment_draw.xml]: @liveHarshit on it

Comment (adityastic) [reply app/src/main/res/layout/fragment_draw.xml]: @liveHarshit yes updated all with the same

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt (hunk 1 above)
Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt]: > There are 3 more unsafe casts like that.

on it

**Comment by adityastic** [comment]
@liveHarshit updaed with the rest, please review

File: app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt (hunk 1 above)
Comment (adityastic) [reply app/src/main/java/org/fossasia/badgemagic/ui/fragments/TextArtFragment.kt]: done, please review

File: app/src/main/res/layout/fragment_draw.xml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+
+        <org.fossasia.badgemagic.ui.custom.DrawBadgeLayout
+            android:id="@+id/draw_layout"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:layout_above="@id/tools_bar"
+            android:layout_alignParentTop="true"
+            android:layout_centerInParent="true"
+            android:layout_marginStart="50dp"
+            android:layout_marginEnd="50dp"
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_draw.xml]: in dimens.xml

File: app/src/main/res/layout/fragment_draw.xml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            app:drawState="@{viewModel.drawModeState}" />
+
+        <LinearLayout
+            android:id="@+id/tools_bar"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_alignParentBottom="true"
+            android:orientation="horizontal"
+            android:padding="10dp">
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_draw.xml]: dimens.xml

File: app/src/main/res/layout/fragment_draw.xml [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+            <LinearLayout
+                android:layout_width="wrap_content"
+                android:layout_height="match_parent"
+                android:layout_margin="10dp"
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_draw.xml]: dimens.xml

File: app/src/main/res/layout/fragment_draw.xml [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+                <TextView
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/erase_text" />
+            </LinearLayout>
+
+            <LinearLayout
+                android:layout_width="wrap_content"
+                android:layout_height="match_parent"
+                android:layout_margin="10dp"
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_draw.xml]: same

File: app/src/main/res/layout/fragment_draw.xml [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+                <TextView
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/reset" />
+            </LinearLayout>
+
+            <LinearLayout
+                android:layout_width="wrap_content"
+                android:layout_height="match_parent"
+                android:layout_margin="10dp"
```
Comment (liveHarshit) [inline app/src/main/res/layout/fragment_draw.xml]: same

File: app/src/main/res/layout/fragment_draw.xml (hunk 4 above)
Comment (adityastic) [reply app/src/main/res/layout/fragment_draw.xml]: updated, @liveHarshit review

**Comment by adityastic** [comment]
@mariobehling @iamareebjamal peer review done

**Comment by adityastic** [comment]
@mariobehling ready to merge

**Pull request merged**